src/components/sbb-option/sbb-option.tsx

```diff
@@ -0,0 +1,196 @@
+import {
+  Component,
+  ComponentInterface,
+  Element,
+  Event,
+  EventEmitter,
+  h,
+  Host,
+  JSX,
+  Listen,
+  Method,
+  Prop,
+  State,
+} from '@stencil/core';
+import { assignId } from '../../global/helpers/assign-id';
+import {
+  createNamedSlotState,
+  queryAndObserveNamedSlotState,
+  queryNamedSlotState,
+} from '../../global/helpers/observe-named-slot-changes';
+import { SbbOptionSelectionChange } from '../sbb-autocomplete/sbb-autocomplete.custom';
+
+let nextId = 0;
+
+/**
+ * @slot unnamed - Use this to provide the option label.
+ * @slot icon - Use this slot to provide an icon. If `icon-name` is set, a sbb-icon will be used.
+ */
+@Component({
+  shadow: true,
+  styleUrl: 'sbb-option.scss',
+  tag: 'sbb-option',
+})
+export class SbbOption implements ComponentInterface {
+  /** Value of the option. */
+  @Prop() public value?: string;
+
+  /**
+   * The icon name we want to use, choose from the small icon variants
+   * from the ui-icons category from here
+   * https://lyne.sbb.ch/tokens/icons/.
+   */
+  @Prop() public iconName?: string;
+
+  /** Whether the icon space is preserved when no icon is set. */
+  @Prop({ reflect: true }) public preserveIconSpace: boolean;
+
+  /** Whether the option is currently active. */
+  @Prop({ reflect: true }) public active?: boolean;
+
+  /** Whether the option is disabled. TBI: missing disabled style, will be implemented with the select component. */
+  @Prop() public disabled?: boolean;
+
+  /** The portion of the highlighted label. */
+  @Prop() public highlightString: string;
+
+  /** Disable the highlight of the label. */
+  @Prop({ reflect: true }) public disableLabelHighlight: boolean;
```

Comment:
This has a similar reason to one of the next discussions. Don't know why but I started commenting from the bottom :|

> When I write components for a broad number of users, I always expect them to have situations that we'd never think of. With this in mind, I would always go for the freest solution.

As of now, the highlight system works only if the user slots a simple text node in an option
`<sbb-option> Highlightable caption </sbb-option>`

E.g of a not working scenario
`<sbb-option> <span> Not highlightable caption </span> </sbb-option>`

Since we are letting the end-user free to customize the option template, there might be cases where the automatic highlight might not be desired or might be inconsistent.
`<sbb-option> <span> Not highlightable </span> highlightable </sbb-option>`

As of now, I would surely remove the 'reflect: true' but If you think it is improbable to happen we can remove the prop

Reply:
Sorry to jump into the conversation 😄 To share my opinion: I would document the scenarios Tommaso described above and remove the `disableLabelHighlight` property. The user is free to insert any content, but should also take into account the specification and behaviour of the component; if he wants nothing to be highlighted, he can nest everything in a span (which would still be wrong as it would break the design).

Reply:
In this specific case, Marco's advice seems a good compromise. 

Still, in general, limiting the API of a component to what we expect will be its use is something that might backfire on us.

Having been to the other side of using a company design system, the answer to a small exception in a use case not supported by the library was: "We can't use it. We need to reimplement it". And once the component is "forked", it is hard to come back on the right track.